R/install_reqs.R

```diff
@@ -162,13 +162,13 @@ install_reqs = function(reqs, dryrun, verbose = dryrun,
 
   # Install git
   failures = failures +
-    install_special_req(reqs$git, GIT_REPLACE, remotes::install_git, dryrun, verbose) +
-    install_special_req(reqs$svn, SVN_EXT_REP, remotes::install_svn, dryrun, verbose) +
-    install_special_req(reqs$bioc, BIO_EXT_REP, remotes::install_bioc, dryrun, verbose) +
-    install_special_req(reqs$url, "", remotes::install_url, dryrun, verbose) +
-    install_special_req(reqs$local, "", remotes::install_local, dryrun, verbose) +
-    install_unversioned(reqs$unversioned, installed, dryrun, verbose, repo) +
-    install_versioned(reqs$versioned, installed, dryrun, verbose, repo)
+    count_failures(reqs$git, install_special_req, GIT_REPLACE, remotes::install_git, dryrun, verbose) +
+    count_failures(reqs$svn, install_special_req, SVN_EXT_REP, remotes::install_svn, dryrun, verbose) +
+    count_failures(reqs$bioc, install_special_req, BIO_EXT_REP, remotes::install_bioc, dryrun, verbose) +
+    count_failures(reqs$url, install_special_req, "", remotes::install_url, dryrun, verbose) +
+    count_failures(reqs$local, install_special_req, "", remotes::install_local, dryrun, verbose) +
+    count_failures(reqs$unversioned, install_unversioned, installed, dryrun, verbose, repo) +
+    count_failures(reqs$versioned, install_versioned, installed, dryrun, verbose, repo)
```

Comment:
really like this rewrite to include the explicitly named count failures function 🎉